### PR TITLE
Ft resource limits

### PR DESCRIPTION
**Pull request by 123MwanjeMike**
**_WHAT DOES THIS PR DO?_**
- Adds resource limits to all resources running on deployment clusters
- Adds horizontal pod scalers for all deployments

**_WHAT ISSUES ARE RELATED TO THIS PR?_**
- Jira cards
    - [PLAT-1147](https://airqoteam.atlassian.net/browse/PLAT-1147?atlOrigin=eyJpIjoiYTJjMGI0YjY0NDU3NDhmMGFmMjQ0OTRlNDBkNzMzYjQiLCJwIjoiaiJ9)
- GitHub issues
    - Closes #<enter GitHub issue number here>

**_HOW DO I TEST OUT THIS PR?_**
- Static code analysis

**_WHICH ENDPOINTS SHOULD BE READY FOR TESTING?:_** 
- n/a

**_IS THERE ANY JENKINS CONSOLE LINK FOR CODE COVERAGE AND BUILD INFO?_**
- n/a

**_ARE THERE ANY RELATED PRs?_**
[This frontend PR](https://github.com/airqo-platform/AirQo-frontend/pull/629)

### CODE REVIEW

File: k8s/calibrate/prod-calibrate-api.yaml [hunk 1]
```diff
@@ -28,11 +28,11 @@ spec:
...
           resources:
             requests:
-              memory: 256Mi
-              cpu: "500m"
+              memory: 100Mi
+              cpu: 50m
             limits:
-              memory: 1Gi
-              cpu: "2000m"
+              memory: 500Mi
```
Comment (123MwanjeMike) [inline k8s/calibrate/prod-calibrate-api.yaml:34]: Resource limits. They are the same for all deployments

File: k8s/channel-data/stage-airqo-update-channel-data-job.yaml [hunk 2]
```diff
@@ -26,7 +26,13 @@ spec:
             - name: stage-update-channel-data-job
               image: us.gcr.io/airqo-250220/airqo-stage-update-channel-data-job:latest
               imagePullPolicy: Always
-              resources: {}
+              resources:
+                requests:
+                  cpu: 20m
+                  memory: 50Mi
+                limits:
+                  cpu: 2000m
```
Comment (123MwanjeMike) [inline k8s/channel-data/stage-airqo-update-channel-data-job.yaml:34]: Resource limits for all cronjobs.

File: k8s/data-mgt/stage-data-mgt-api.yaml [hunk 3]
```diff
@@ -35,7 +35,13 @@ spec:
           envFrom:
             - configMapRef:
                 name: data-mgt-api-config
-          resources: {}
+          resources:
+            requests:
+              cpu: 100m
```
Comment (123MwanjeMike) [inline k8s/data-mgt/stage-data-mgt-api.yaml:40]: Default allocated resources for all deployment pods.

File: k8s/data-mgt/stage-data-mgt-api.yaml [hunk 4]
```diff
@@ -53,3 +59,23 @@ spec:
...
+    name: airqo-stage-data-mgt-api
+  minReplicas: 1
+  maxReplicas: 2
+  metrics:
+  - type: Resource
+    resource:
+      name: cpu
+      target:
+        type: Utilization
+        averageUtilization: 80
```
Comment (123MwanjeMike) [inline k8s/data-mgt/stage-data-mgt-api.yaml:81]: Percentage of allocated cpu on a pod on which to depend when auto scaling

**Review by 123MwanjeMike — COMMENTED** [review]
A few guiding comments

File: k8s/view-api/stage-view-api.yaml [hunk 5]
```diff
@@ -26,63 +26,64 @@ spec:
...
+          # readinessProbe:
+          #   initialDelaySeconds: 30
+          #   httpGet:
+          #     port: 8080
+          #     path: api/v1/view/actuator/health/readiness
+          # livenessProbe:
+          #   initialDelaySeconds: 30
+          #   httpGet:
+          #     port: 8080
+          #     path: api/v1/view/actuator/health/liveness
```
Comment (NoahNsimbe) [inline k8s/view-api/stage-view-api.yaml:51]: Hi @123MwanjeMike any reason for removing the readiness and liveness probes??

Comment (123MwanjeMike) [reply k8s/view-api/stage-view-api.yaml:51]: They were stopping the pods from starting. I guess they were not picking the right information. Removing them seemed to solve the issue. Perhaps you could double check to be sure.

Comment (NoahNsimbe) [reply k8s/view-api/stage-view-api.yaml:51]: https://platform.airqo.net/api/v1/view/actuator/health/liveness
https://platform.airqo.net/api/v1/view/actuator/health/readiness

We need to put them back. They are very useful when performing updates and many other usecases. If the pod is not starting, then something is not right and needs to be fixed. If we remove them, we shall never know that the pod has issues.

https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/
https://developers.redhat.com/blog/2020/11/10/you-probably-need-liveness-and-readiness-probes
https://blog.colinbreck.com/kubernetes-liveness-and-readiness-probes-how-to-avoid-shooting-yourself-in-the-foot/

**Review by NoahNsimbe — COMMENTED** [review]
Hi @123MwanjeMike could you confirm that the total cpu and memory requests are less than or equal to the total cpu and memory of the cluster. Otherwise, some pods might never be scheduled.

File: k8s/view-api/stage-view-api.yaml (hunk 5 above)
Comment (123MwanjeMike) [reply k8s/view-api/stage-view-api.yaml]: Thanks @NoahNsimbe 
I've added the probes back

**Comment by 123MwanjeMike** [comment]
> Hi @123MwanjeMike could you confirm that the total cpu and memory requests are less than or equal to the total cpu and memory of the cluster. Otherwise, some pods might never be scheduled.

Yes @NoahNsimbe. The total allocated resources are less than the cluster resources

**Review by NoahNsimbe — APPROVED** [review]
LGTM. 
We can monitor performance on staging before moving to production @123MwanjeMike

**Pull request merged**